src/main/java/com/pubnub/api/endpoints/access/GrantToken.java

```diff
@@ -0,0 +1,130 @@
+    private List<Space> spaceList;
+
+    public GrantToken(PubNub pubnub, TelemetryManager telemetryManager, RetrofitManager retrofit) {
+        super(pubnub, telemetryManager, retrofit);
+        userList = new ArrayList<>();
+        spaceList = new ArrayList<>();
+    }
+
+    @Override
+    protected List<String> getAffectedChannels() {
+        // generate a list of channels when they become supported in PAMv3
+        return null;
+    }
+
+    @Override
+    protected List<String> getAffectedChannelGroups() {
+        // generate a list of groups when they become supported in PAMv3
+        return null;
+    }
+
+    @Override
+    protected void validateParams() throws PubNubException {
+        if (this.getPubnub().getConfiguration().getSecretKey() == null || this.getPubnub()
+                .getConfiguration()
+                .getSecretKey()
+                .isEmpty()) {
+            throw PubNubException.builder().pubnubError(PubNubErrorBuilder.PNERROBJ_SECRET_KEY_MISSING).build();
+        }
+        if (this.getPubnub().getConfiguration().getSubscribeKey() == null || this.getPubnub()
+                .getConfiguration()
+                .getSubscribeKey()
+                .isEmpty()) {
+            throw PubNubException.builder().pubnubError(PubNubErrorBuilder.PNERROBJ_SUBSCRIBE_KEY_MISSING).build();
+        }
+        if ((this.userList == null || this.userList.isEmpty())
+                && (this.spaceList == null || this.spaceList.isEmpty())) {
+            throw PubNubException.builder()
+                    .pubnubError(PubNubErrorBuilder.PNERROBJ_RESOURCES_MISSING)
+                    .build();
+        }
+        if (this.ttl == null) {
+            throw PubNubException.builder()
+                    .pubnubError(PubNubErrorBuilder.PNERROBJ_TTL_MISSING)
+                    .build();
+        }
+    }
+
+    @Override
+    protected Call<JsonObject> doWork(Map<String, String> queryParams) throws PubNubException {
+        JsonObject requestBody = GrantTokenRequestBody.builder()
+                .pubNub(this.getPubnub())
+                .ttl(ttl)
+                .spaces(spaceList)
+                .users(userList)
+                .meta(meta)
+                .build()
+                .assemble();
+
+        return this.getRetrofit()
+                .getAccessManagerService()
+                .grantToken(this.getPubnub().getConfiguration().getSubscribeKey(), requestBody, queryParams);
+    }
+
+    @Override
+    protected PNGrantTokenResult createResponse(Response<JsonObject> input) throws PubNubException {
+        PNGrantTokenResult.PNGrantTokenResultBuilder builder = PNGrantTokenResult.builder();
+
+        if (input.body() != null) {
+            builder.token(input.body().getAsJsonObject("data").get("token").getAsString());
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    protected PNOperationType getOperationType() {
+        return PNOperationType.PNAccessManagerGranToken;
```

Comment:
This seems like a possible typo? `PNAccessManagerGranToken` <-- Missing the "t" in `Grant`